Removed the `exists` function.

arpd opened the pull request:
As described in #43 :
  `fopen` will fail if we attempt to open a non-existent file, or one
  with permissions that prevent us reading it.

dafky2000 commented:
If we are getting rid of that function, simply delete the unit test file, `spec/template_functions_exists.c`. No need to test if the function no longer exists :)

arpd commented:
@dafky2000 Done; I was getting valgrind failures due to `make test` not rebuilding.

dafky2000 commented:
Thank you for tidying that up! It appears that even if building fails, valgrind still attempts to run. I will try to address that.

I am closing this pull request, could you resubmit your pull request to the `develop` branch? 

Thank you!

arpd committed:
Removed the `exists` function.

As described in #43 :
  `fopen` will fail if we attempt to open a non-existent file, or one
  with permissions that prevent us reading it.
